test/integration/housekeeping_test.go

```diff
@@ -0,0 +1,160 @@
+	"context"
+	"fmt"
+	"log"
+	"sort"
+	"testing"
+
+	gotime "time"
+
+	monkey "github.com/undefinedlabs/go-mpatch"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/yorkie-team/yorkie/api/types"
+	"github.com/yorkie-team/yorkie/server/backend/database"
+	"github.com/yorkie-team/yorkie/server/backend/database/mongo"
+	"github.com/yorkie-team/yorkie/server/backend/housekeeping"
+	"github.com/yorkie-team/yorkie/server/backend/sync/memory"
+	"github.com/yorkie-team/yorkie/test/helper"
+)
+
+const (
+	dummyOwnerID              = types.ID("000000000000000000000003")
+	otherOwnerID              = types.ID("000000000000000000000004")
+	clientDeactivateThreshold = "23h"
+)
+
+func setupTest(t *testing.T) *mongo.Client {
+	config := &mongo.Config{
+		ConnectionTimeout: "5s",
+		ConnectionURI:     "mongodb://localhost:27017",
+		YorkieDatabase:    helper.TestDBName() + "-integration",
+		PingTimeout:       "5s",
+	}
+	assert.NoError(t, config.Validate())
+
+	cli, err := mongo.Dial(config)
+	assert.NoError(t, err)
+
+	return cli
+}
+
+func TestHousekeeping(t *testing.T) {
+	config := helper.TestConfig()
+	db := setupTest(t)
+
+	projects := createProjects(t, db)
+
+	coordinator := memory.NewCoordinator(nil)
+
+	h, err := housekeeping.New(config.Housekeeping, db, coordinator)
+	assert.NoError(t, err)
+
+	t.Run("`FindDeactivateCandidates` should return correct last projectID", func(t *testing.T) {
+		ctx := context.Background()
+
+		fetchSize := 3
+		lastProjectID := database.DefaultProjectID
+
+		for i := 0; i < len(projects)/fetchSize; i++ {
+			lastProjectID, _, err = h.FindDeactivateCandidates(
+				ctx,
+				0,
+				fetchSize,
+				lastProjectID,
+			)
+			assert.NoError(t, err)
+			assert.Equal(t, projects[((i+1)*fetchSize)-1].ID, lastProjectID)
+		}
+
+		lastProjectID, _, err = h.FindDeactivateCandidates(
+			ctx,
+			0,
+			fetchSize,
+			lastProjectID,
+		)
+		assert.NoError(t, err)
+		assert.Equal(t, projects[fetchSize-(len(projects)%3)-1].ID, lastProjectID)
+	})
+
+	t.Run("`FindDeactivateCandidates` should return correct clients", func(t *testing.T) {
+		ctx := context.Background()
+
+		yesterday := gotime.Now().Add(-24 * gotime.Hour)
+		patch, err := monkey.PatchMethod(gotime.Now, func() gotime.Time { return yesterday })
+		if err != nil {
+			log.Fatal(err)
+		}
+		clientA, err := db.ActivateClient(ctx, projects[0].ID, fmt.Sprintf("%s-A", t.Name()))
+		assert.NoError(t, err)
+		clientB, err := db.ActivateClient(ctx, projects[0].ID, fmt.Sprintf("%s-B", t.Name()))
+		assert.NoError(t, err)
+		err = patch.Unpatch()
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		clientC, err := db.ActivateClient(ctx, projects[0].ID, fmt.Sprintf("%s-C", t.Name()))
+		assert.NoError(t, err)
+
+		_, candidates, err := h.FindDeactivateCandidates(
+			ctx,
+			10,
+			10,
+			database.DefaultProjectID,
+		)
+
+		assert.NoError(t, err)
+		assert.Len(t, candidates, 2)
+		assert.Equal(t, candidates[0].ID, clientA.ID)
+		assert.Contains(t, candidates[1].ID, clientB.ID)
```

Comment:
How about using only `assert.Equal(t, candidates[1].ID, clientB.ID)` or `assert.Contains(t, candidates, clientA)` to keep consistency?